helpers/tempesta.py

```diff
@@ -33,16 +33,11 @@ def version():
     global tfw_version
     if tfw_version:
         return tfw_version
-    version_header = ''.join([os.path.dirname(os.path.realpath(__file__)),
-                              '/../../../tempesta_fw.h'])
-    with open(version_header, 'r') as header:
-        read_data = header.read()
-        m = re.search(r'#define TFW_VERSION\s+"([0-9a-z.-]+)"', read_data)
-        error.assertTrue(m)
-        tfw_version = m.group(1)
-        return tfw_version
-    error.bug()
-
+    hdr_filename = remote.tempesta.workdir+"/tempesta_fw/tempesta_fw.h"
+    parse_cmd = r"grep TFW_VERSION | awk -F '[\" ]' '{printf $3}'"
+    cmd = "cat %s | %s" % (hdr_filename, parse_cmd)
+    version, _ = remote.tempesta.run_cmd(cmd=cmd)
+    return version
```

Comment:
In old implementation version was cached into `tfw_version` variable, so the regex search was called only once. In new implementation cached variable is not updated, so regex search will be called on every `version()` call. Version string caching has a great impact to tests performance:  the `version()` call is used to build every expected response for deproxy tests.

Also need assert that `version` is not empty.

Reply:
fixed